x-pack/legacy/plugins/ml/public/jobs/new_job_new/common/job_validator/util.ts

```diff
@@ -151,6 +151,20 @@ export function populateValidationMessages(
 
     basicValidations.bucketSpan.message = msg;
   }
+
+  if (validationResults.contains('query_empty')) {
+    basicValidations.query.valid = false;
+    const msg = i18n.translate('xpack.ml.newJob.wizard.validateJob.queryCannotBeEmpty', {
+      defaultMessage: 'Datafeed query cannot be empty.',
+    });
+    basicValidations.query.message = msg;
+  } else if (validationResults.contains('query_invalid')) {
+    basicValidations.query.valid = false;
+    const msg = i18n.translate('xpack.ml.newJob.wizard.validateJob.queryIsInvalidJSON', {
```

Comment:
Nit - would something like `queryIsInvalidElasticsearchQuery` be better than `queryIsInvalidJSON` for this i18n message?